packages/react-data-grid/src/HeaderCell.tsx

```diff
@@ -0,0 +1,115 @@
+import React from 'react';
+import classNames from 'classnames';
+import { isElement } from 'react-is';
+import Draggable from './Draggable';
+import { isFrozen } from './ColumnUtils';
+import { HeaderRowType } from './common/enums';
+import { Column } from './common/types';
+
+function SimpleCellRenderer({ column }: { column: Column }) {
+  const headerText = column.rowType === HeaderRowType.HEADER ? column.name : '';
+  return <div>{headerText}</div>;
+}
+
+interface Props {
+  renderer?: React.ReactElement | React.FunctionComponent<{ column: Column }>;
+  column: Column;
+  height: number;
+  onResize(column: Column, width: number): void;
+  onResizeEnd(column: Column, width: number): void;
+  onHeaderDrop(): void;
+  draggableHeaderCell: React.ComponentType<{ column: Column; onHeaderDrop(): void }>;
+  className?: string;
+}
+
+export default class HeaderCell extends React.Component<Props> {
+  headerCell = React.createRef<HTMLDivElement>();
+
+  onDrag = (x: number) => {
+    const { onResize } = this.props;
+    if (onResize) {
+      const width = this.getWidthFromMouseEvent(x);
+      if (width > 0) {
+        onResize(this.props.column, width);
+      }
+    }
+  };
+
+  onDragEnd = (x: number) => {
+    const width = this.getWidthFromMouseEvent(x);
+    this.props.onResizeEnd(this.props.column, width);
+  };
+
+  getWidthFromMouseEvent(x: number): number {
+    const left = this.headerCell.current!.getBoundingClientRect().left;
```

Comment:
We should probably check if `this.this.headerCell.current` is not null
https://github.com/adazzle/react-data-grid/pull/1564/files#diff-58d2c424802f17b3bf163a3ae5b4469fR62

Reply:
This method is called when you're in the middle of dragging, `headerCell.current` cannot possibly be `null` at this point.